testng-core-api/src/main/java/org/testng/ITestResult.java

```diff
@@ -101,47 +102,49 @@ public interface ITestResult extends IAttributes, Comparable<ITestResult> {
   void setWasRetried(boolean wasRetried);
 
   /**
-   * @return - The list of either upstream method(s) or configuration method(s) whose failure led to
-   *     the current method being skipped. An empty list is returned when the current method is not
-   *     a skipped method.
+   * Returns the list of either upstream method(s) or configuration method(s) whose failure led to
+   * the current method being skipped. An empty list is returned when the current method is not a
+   * skipped method.
    */
   default List<ITestNGMethod> getSkipCausedBy() {
     return Collections.emptyList();
   }
 
   /**
-   * @return - A unique id for the current JVM that represents a unique way of identifying a
-   *     specific test method's result.
+   * Returns a unique id for the current JVM that represents a unique way of identifying a specific
+   * test method's result.
    */
   String id();
 
   /**
-   * @return - <code>true</code> if the current test result is either {@link ITestResult#STARTED} or
-   *     {@link ITestResult#CREATED}
+   * Returns <code>true</code> if the current test result is either {@link ITestResult#STARTED} or
+   * {@link ITestResult#CREATED}
    */
   default boolean isNotRunning() {
     return getStatus() == STARTED || getStatus() == CREATED;
   }
 
   /**
-   * @return - A list of all user facing statuses viz.,
-   *     <ul>
-   *       <li>{@link ITestResult#SUCCESS}
-   *       <li>{@link ITestResult#SUCCESS_PERCENTAGE_FAILURE}
-   *       <li>{@link ITestResult#FAILURE}
-   *       <li>{@link ITestResult#SKIP}
-   *     </ul>
+   * Returns a list of all user facing statuses viz.,
+   *
+   * <ul>
+   *   <li>{@link ITestResult#SUCCESS}
+   *   <li>{@link ITestResult#SUCCESS_PERCENTAGE_FAILURE}
+   *   <li>{@link ITestResult#FAILURE}
+   *   <li>{@link ITestResult#SKIP}
+   * </ul>
    */
   static List<String> finalStatuses() {
     return Arrays.asList("SUCCESS", "FAILURE", "SKIP", "SUCCESS_PERCENTAGE_FAILURE");
   }
 
   /**
+   * Returns <code>true</code> if the test failure was due to a timeout.
+   *
    * @param result - The test result of a method
-   * @return - <code>true</code> if the test failure was due to a timeout.
    */
   static boolean wasFailureDueToTimeout(ITestResult result) {
-    Throwable cause = result.getThrowable();
+    @Nullable Throwable cause = result.getThrowable();
```

Comment:
Technically speaking, `@Nullable` should be automatically inferred for local variables, so this could probably be omitted

```suggestion
    Throwable cause = result.getThrowable();
```

Reply:
I suppose you're right.

It means I should not use `TypeUseLocation.ALL` in `@DefaultQualifier(value = NonNull.class, locations = TypeUseLocation.ALL)`
Any other option I should exclude? https://checkerframework.org/api/org/checkerframework/framework/qual/TypeUseLocation.html